### PR TITLE
Add `CaseConverter__` values to `CaseConverterEnum`, now in PR form

### LINKED ISSUE #62: Add `caseConverter__` values to `CaseConverterEnum`
The thing that really caught my eye with this lib is how it lets you write perfectly valid, type-safe files in typescript if you only restrict the use of case conversion to strings and comments. Unfortunately if you want to use case converters there are things you simply can't do. 

For example, there is no way to make this snippet valid TS code.

```ts
// Given `__resourceName__ `is PascalCalse
export const __resourceName__(constantCase) _KEY = "__resourceName__(snakeCase)_exported_type";
const get__resourceName__ = (): Promise<ResultFor<__resourceName__(snakeCase)_exported_type>> => {
  get(__resourceName__(constantCase) _KEY)
}

```

This could be alleviated by allowing the case conversion tags to be underscored. In that case the above would look as follows:

```ts
// Given `__resourceName__ `is PascalCalse
export const __resourceName__constantCase___KEY = "__resourceName__snakeCase___exported_type";
export const get__resourceName__ = (): Promise<ResultFor<__resourceName__snakeCase___exported_type>> => {
  get(__resourceName__constantCase___KEY)
}
```

This ends up being totally valid TS, and at worst maybe need a few lint rules disabled. There's also no reason why the two couldn't coexist, so there shouldn't be any risk to existing users. 

I've already tested that it works [here](https://github.com/TikiTDO/generate-template-files/pull/1/files). If you'd like I can merge this code into my other PR.

### PR DESCRIPTION
**Pull request by TikiTDO**
Closes #62

Adds another variant to the case converter enum in order to facilitate syntactically valid JS and TS in templates.

### CODE REVIEW

File: examples/tools/templates/misc/__resourceName__pascalCase__.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  }
+};
+
+/**
+ * The __resourceName__(titleCase) fetcher
+ */
+export const get__resourceName__ = async (): Promise<__resourceName__pascalCase__Data> => {
+  const __resourceName__camelCase__ = fetchData(__resourceName__snakeCase___data.key);
+  return __resourceName__camelCase__;
+};
```
Comment (TikiTDO) [inline examples/tools/templates/misc/__resourceName__pascalCase__.ts]:
```ts
// src/example/UserProfile.ts
import { user_profile_data } from "./UserProfileData"

console.assert(user_profile_data.key === "user_profile");

export const USER_PROFILE_KEY = user_profile_data.key;
type UserProfileData = typeof user_profile_data

/**
 *
 */
const fetchData = (_key: "user_profile"): UserProfileData => {
  // Implement fetch logic here
  const result: UserProfileData["result"] = {}

  return {
    key: USER_PROFILE_KEY,
    result,
  }
};

/**
 * The User Profile fetcher
 */
export const getUserProfile = async (): Promise<UserProfileData> => {
  const userProfile = fetchData(user_profile_data.key);
  return userProfile;
};

```

File: examples/tools/templates/misc/__resourceName__pascalCase__Data.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+// src/__scope__camelCase__/__resourceName__(pascalCase)Data.ts
+export const __resourceName__snakeCase___data = {
+  key: "__resourceName__(snakeCase)",
+  result: {
+    // Add values here
+  }
+} as const
```
Comment (TikiTDO) [inline examples/tools/templates/misc/__resourceName__pascalCase__Data.ts]:
```ts
// src/example/UserProfileData.ts
export const user_profile_data = {
  key: "user_profile",
  result: {
    // Add values here
  }
} as const
```

File: src/constants/CaseConverterEnum.ts [hunk 3]
```diff
@@ -18,12 +18,13 @@ enum CaseConverterEnum {
    * ```
    */
   None = '(noCase)',
+  NoneUnderscore = 'noCase__',
```
Comment (codeBelt) [inline src/constants/CaseConverterEnum.ts]: I am kinda thinking PascalCase would be nicer for this new ones?

```suggestion
  NoneUnderscore = 'NoCase__',
```

```js
__replacerSlot__NoCase__
```

Comment (TikiTDO) [reply src/constants/CaseConverterEnum.ts]: Sounds good, think I got them all

**Pull request merged**